include/orc/parse_file.hpp

```diff
@@ -226,19 +226,25 @@ constexpr std::decay_t<T> copy(T&& value) noexcept(noexcept(std::decay_t<T>{
 
 /**************************************************************************************************/
 
-using register_dies_callback = std::function<void(dies)>;
-using do_work_callback = std::function<void(std::function<void()>)>;
-using empool_callback = std::function<pool_string(std::string_view)>;
+enum class macho_reader_mode {
+    invalid,
+    register_dies,
+    derive_dylibs,
+    odrv_reporting,
+};
+
+struct macho_params {
```

Comment:
`macho_params` surrenders to the fact that all this file scanning, in the end, is Mach-O based. At one point I was writing ORC more from the perspective that it shouldn't matter what the ABI is and the ORC file scanners should be more generic. Perhaps I was thinking we'd be able to incorporate Windows more easily at some point in the future? Who know. At any rate, the added genericity resulted in passing around the `callbacks` block everywhere, as some kind of pseudo-delegate pattern that wasn't very well thought through. I have since removed it and replaced it with `macho_params`. This is a parameter block that is created at the top level `orc_process` call and is sent into the file scanning system to tell the Mach-O parser exactly what it should be looking out for. Hopefully this setup is more clear than the `callbacks` mechansim we had before.

Reply:
I like the idea of windows support, but at this point, I think that would be more about abstracting out the ORC core from the parser. ORC would begin to look more like a multi-platform app, where you have the "core" parts and the "platform/compiler" parts. At that point, we would need to abstract the macho params.

But it's all very theoretical - especially w/o a MSVC implementation, so I think the approach of streamlining what we actually have is the correct one.

Reply:
Agreed.